lib/3scale/backend/worker_async.rb

```diff
@@ -73,6 +73,10 @@ def process_all
           break if @shutdown
 
           semaphore.async { perform(job) }
+
+          # Wait for the last `@max_concurrent_jobs` tasks to finish and clean the barrier.
+          # Otherwise it produces a memory leak. More info: https://github.com/3scale/apisonator/pull/376
```

Comment:
```suggestion
          # Clean-up tasks inside barrier regularly, otherwise they accumulate throughout the worker lifetime
          # and never GCed, eventually exhausting the whole available memory. Moreover the array keeping
          # track of tasks in the barrier grows indefinitely too occupying memory and reducing performance.
```

Something along these lines I think summarizes it. Or no?